go.mod

```diff
@@ -1,29 +1,26 @@
 module github.com/operator-framework/kubectl-operator
 
-go 1.23.0
-
-toolchain go1.23.4
+go 1.23.4
```

Comment:
Since it is as we do in oper-controller: https://github.com/operator-framework/operator-controller/blob/main/go.mod#L3

I am fine with the change 👍